apps/meteor/server/email/IMAPInterceptor.ts

```diff
@@ -19,27 +20,36 @@ export declare interface IMAPInterceptor {
 export class IMAPInterceptor extends EventEmitter {
 	private imap: IMAP;
 
+	private initialBackoffDurationMS = 30000;
+
+	private backoff: NodeJS.Timeout;
+
+	private retries = 0;
+
 	constructor(
 		imapConfig: IMAP.Config,
 		private options: IMAPOptions = {
 			deleteAfterRead: false,
-			filter: ['UNSEEN'],
+			filter: ['!SEEN'],
 			markSeen: true,
+			maxRetries: 10,
 		},
 	) {
 		super();
 
 		this.imap = new IMAP({
-			connTimeout: 30000,
+			connTimeout: 300000,
 			keepalive: true,
 			...imapConfig,
 		});
 
 		// On successfully connected.
 		this.imap.on('ready', () => {
 			if (this.imap.state !== 'disconnected') {
+				clearTimeout(this.backoff);
```

Comment:
```suggestion
				clearTimeout(this.backoff);
				this.retries = 0;
```
Think you missed resetting this counter once the connection was successful :sweat_smile: